File: fix_sidebar.js
const fs = require('fs');
const content = fs.readFileSync('src/components/Sidebar.tsx', 'utf-8');

// The botched replacement inserted the button inside the leaderboard map.
// I will just replace the whole leaderboard div back to its correct state
// and insert the button inside the `segmentRange && activities[0]` block.

let newContent = content.replace(/<div className="space-y-2">\s*<label className="text-xs text-gray-400">Trim Segment Start \/ End<\/label>[\s\S]*?{useAppStore\.getState\(\)\.segmentResults\.sort\(\(a, b\) => a\.duration - b\.duration\)\.map\(\(res, i\) => {[\s\S]*?}\)}/,
`
                    <div className="space-y-2">
                      <label className="text-xs text-gray-400">Trim Segment Start / End</label>
                      <div className="flex gap-2">
                        <input 
                          type="range" 
                          min={seg.startIndex} 
                          max={segmentRange[1] - 1} 
                          value={segmentRange[0]} 
                          onChange={(e) => setSegmentRange([parseInt(e.target.value), segmentRange[1]])}
                          className="flex-1 h-1 bg-gray-700 rounded-lg appearance-none accent-purple-500"
                        />
                        <input 
                          type="range" 
                          min={segmentRange[0] + 1} 
                          max={seg.endIndex} 
                          value={segmentRange[1]} 
                          onChange={(e) => setSegmentRange([segmentRange[0], parseInt(e.target.value)])}
                          className="flex-1 h-1 bg-gray-700 rounded-lg appearance-none accent-purple-500"
                        />
                      </div>
                      <button
                        onClick={(e) => {
                          e.stopPropagation();
                          useAppStore.getState().extractSegment();
                        }}
                        className="w-full mt-2 py-1.5 bg-gray-700 hover:bg-gray-600 text-gray-200 text-xs font-medium rounded transition-colors"
                      >
                        Save as New Segment
                      </button>
                    </div>
                  )}

                  {/* Leaderboard for active segment */}
                  <div className="mt-4 pt-4 border-t border-gray-700 space-y-2">
                    <h4 className="text-xs text-gray-400 uppercase">Leaderboard</h4>
                    {useAppStore.getState().segmentResults.sort((a, b) => a.duration - b.duration).map((res, i) => {
`);
// wait, the regex replacement might be tricky. Let's just use simple search and replace string.

fs.writeFileSync('src/components/Sidebar.tsx', newContent);

File: fix_sidebar.py
import re

with open('src/components/Sidebar.tsx', 'r') as f:
    content = f.read()

# Add Shield icon
content = content.replace(
    "Trash2 } from 'lucide-react';",
    "Trash2, Shield } from 'lucide-react';"
)

# Extract privacyMode from store
content = content.replace(
    "segmentRange, setSegmentRange, setAppMode, extractSegment, deleteSegment",
    "segmentRange, setSegmentRange, setAppMode, extractSegment, deleteSegment, privacyMode, setPrivacyMode"
)

# Insert toggle UI right before activities map
old_activities = """      <div className="flex-1 overflow-y-auto p-4 space-y-4">
        {activities.map(activity => ("""

new_activities = """      <div className="flex-1 overflow-y-auto p-4 space-y-4">
        {appMode === 'event' && activities.length > 1 && detectedSegments.length > 0 && (
          <div className="bg-gray-800 rounded-lg p-3 flex items-center justify-between border border-gray-700">
            <div className="flex items-center gap-2">
              <Shield className={`w-4 h-4 ${privacyMode ? 'text-green-400' : 'text-gray-500'}`} />
              <span className="text-sm font-medium text-gray-200">Meetup Privacy Zone</span>
            </div>
            <button
              onClick={() => setPrivacyMode(!privacyMode)}
              className={`relative inline-flex h-5 w-9 items-center rounded-full transition-colors focus:outline-none ${privacyMode ? 'bg-green-500' : 'bg-gray-600'}`}
            >
              <span className={`inline-block h-3 w-3 transform rounded-full bg-white transition-transform ${privacyMode ? 'translate-x-5' : 'translate-x-1'}`} />
            </button>
          </div>
        )}
        
        {activities.map(activity => ("""

content = content.replace(old_activities, new_activities)

with open('src/components/Sidebar.tsx', 'w') as f:
    f.write(content)

